CONTRIBUTING.md

```diff
@@ -104,7 +104,9 @@ and re-running `npm install`.
 Environment specific settings go in the untracked file `config.js`. Copy from one of
```

Comment:
> untracked file

Ah yes, that was why.

I'm sure it's fine.

Reply:
Uh. :sweat_smile:

Reply:
Generally speaking this could be done by having a `local.config.js` which is still ignored, and the build script will read it if present.

esbuild's `inject` option might work.

I'd be fine to leave it and just avoid coming changes accidentally, or you could back it out.

Reply:
You'd still have to explicitly `git <whatever> -f` to accidentally commit changes, no?

Reply:
No, it's easy to do `git add -u` or `git commit -a` on autopilot and make a change. gitignore only keeps the file untracked; once a file is tracked it doesn't get ignored.

There's skip-worktree or assume-unchanged. Not sure exactly, but the file could have a comment up top saying how to instruct git to ignore changes to it.

I'm presuming that most users will use the default and only occasionally switch to a different source.

Reply:
Can you set a CI task that fails if someone adds certain file names? Then if they screw up, it's not a problem.

Other related things are .gitattributes merge behavior (which may only affect conflicts), smudge filters, and the skip-worktree/assume-unchanged mentioned above.

The annoyance with the latter that I kept encountering is that if anyone *else* did it wrong, everyone who did it right had to un-skip, merge, and re-skip. And if anyone forgot to re-skip, it would start the cycle again.

Reply:
I'm not quite sure how to do
`import config from "./config.js";`
but make it be "and if not there, grab it from `config.default.js` ?

Reply:
I got it to work. Pushing up a commit…

Reply:
Things such as git hooks, index bits, attributes, filters, etc. etc. are interesting in principle but they have to be configured separately from cloning, only the simplest of attributes are built-in to git. So I don't think they're ideal for this case.